File: eduparent/backend/src/main/java/com/netcracker/edu/backend/service/ProjectService.java
package com.netcracker.edu.backend.service;

import com.netcracker.edu.backend.model.Project;
import org.springframework.data.domain.Page;

public interface ProjectService {
    Iterable<Project> getAllProjects();
    Project saveProject(Project project);
    Project findProjectById(Long id);
    Project findProjectByCode(String code);
    void deleteProject(Long id);
    Page<Project> getCurrentProjects(Long page);
    Long getNumberOfProjects();
}

File: eduparent/frontend/src/app/service/project/project.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Project} from "../../model/project";

@Injectable({
  providedIn: 'root'
})
export class ProjectService {
  constructor(private http: HttpClient) {
  }

  getProjects(): Observable<Project[]> {
    return this.http.get<Project[]>('/api/projects');
  }

  getProjectByCode(code: string): Observable<Project> {
    return this.http.get<Project>('/api/projects/code/' + code);
  }

  getCurrentProjects(page: number): Observable<any> {
    return this.http.get<any>('/api/projects/page/' + page);
  }

  getNumberOfProjects(): Observable<number>{
    return this.http.get<number>('/api/projects/number');
  }

  getProjectById(id: number): Observable<Project> {
    return this.http.get<Project>('/api/projects/' + id);
  }

  saveProject(project: Project): Observable<Project> {
    return this.http.post<Project>('/api/projects', project);
  }

  deleteProject(projectId: number): Observable<void> {
    return this.http.delete<void>('/api/projects/' + projectId);
  }
}
